tensorflow/compiler/xla/service/gpu/target_util.cc

```diff
@@ -66,6 +71,30 @@ struct TargetIntrinsics GetIntrinsic(TargetIntrinsicID intrin) {
       return {llvm::Intrinsic::nvvm_barrier0,
               llvm::Intrinsic::amdgcn_s_barrier};
     }
+    case TargetIntrinsicID::kBlockDimx: {
+      return {llvm::Intrinsic::nvvm_read_ptx_sreg_ntid_x,
+              [](llvm::IRBuilder<>* b_) -> llvm::CallInst* {
+                return EmitDeviceFunctionCall("__ockl_get_local_size",
+                                              {b_->getInt32(0)}, {U32}, U64, {},
+                                              b_);
+              }};
+    }
+    case TargetIntrinsicID::kBlockDimy: {
+      return {llvm::Intrinsic::nvvm_read_ptx_sreg_ntid_y,
+              [](llvm::IRBuilder<>* b_) -> llvm::CallInst* {
+                return EmitDeviceFunctionCall("__ockl_get_local_size",
+                                              {b_->getInt32(1)}, {U32}, U64, {},
+                                              b_);
+              }};
+    }
+    case TargetIntrinsicID::kBlockDimz: {
+      return {llvm::Intrinsic::nvvm_read_ptx_sreg_ntid_z,
+              [](llvm::IRBuilder<>* b_) -> llvm::CallInst* {
+                return EmitDeviceFunctionCall("__ockl_get_local_size",
+                                              {b_->getInt32(1)}, {U32}, U64, {},
```

Comment:
I believe this should be b_->getInt32(2) rather than (1) which would by "y".

I wish we had test coverage for target_util.cc. However, I will also not ask you to be the first one to write tests here. That ship has sailed.

Reply:
Thanks for catching this. Will fix it.